jina/drivers/segment.py

```diff
@@ -19,24 +20,44 @@ def __init__(
         method: str = 'segment',
         traversal_paths: Tuple[str] = ('r',),
         *args,
-        **kwargs
+        **kwargs,
     ):
         super().__init__(
             executor, method, traversal_paths=traversal_paths, *args, **kwargs
         )
 
     def _apply_all(self, docs: 'DocumentSet', *args, **kwargs):
+        if not docs:
+            self.logger.warning(f'an empty DocumentSet {docs}')
+            return
```

Comment:
Interesting, it is saying that some cases are not covered by tests. Is it correct?

Reply:
wekk right now this case is also covered in the mixin